engine/new_runtime/src/memories/ram_memory.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    locked_memory::LockedMemory,
+    memories::buffer::Buffer,
+    MemoryError::{self, *},
+    ZeroizeOnDrop, DEBUG_MSG,
+};
+use core::{
+    fmt::{self, Debug, Formatter},
+    marker::PhantomData,
+};
+use zeroize::Zeroize;
+
+use serde::{
+    de::{Deserialize, Deserializer, SeqAccess, Visitor},
+    ser::{Serialize, Serializer},
+};
+
+/// Protected ram memory that may be encrypted or not
+/// This is basically a wrapper for the Buffer type, but the usage
+/// is different, buffer type are meant for short lived usage while
+/// RamMemory can store data for longer period of time.
+/// Hence data in RamMemory has to be either encyrpted or protected
+/// behind a scheme
+#[derive(Clone)]
+pub struct RamMemory {
```

Comment:
```suggestion
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RamMemory {
```

No need to manually implement `Debug`, `Serialize` and `Deserialize` below. They are already implemented on `Buffer`, so we can just derive them here.